test/e2e/node/kubelet.go

```diff
@@ -309,52 +301,46 @@ var _ = SIGDescribe("kubelet", func() {
-			ginkgo.It(name, func() {
-				totalPods := itArg.podsPerNode * numNodes
-				ginkgo.By(fmt.Sprintf("Creating a RC of %d pods and wait until all pods of this RC are running", totalPods))
-				rcName := fmt.Sprintf("cleanup%d-%s", totalPods, string(uuid.NewUUID()))
-
-				err := framework.RunRC(testutils.RCConfig{
-					Client:       f.ClientSet,
-					Name:         rcName,
-					Namespace:    f.Namespace.Name,
-					Image:        imageutils.GetPauseImageName(),
-					Replicas:     totalPods,
-					NodeSelector: nodeLabels,
-				})
-				framework.ExpectNoError(err)
-				// Perform a sanity check so that we know all desired pods are
-				// running on the nodes according to kubelet. The timeout is set to
-				// only 30 seconds here because framework.RunRC already waited for all pods to
-				// transition to the running status.
-				err = waitTillNPodsRunningOnNodes(f.ClientSet, nodeNames, rcName, ns, totalPods, time.Second*30)
-				framework.ExpectNoError(err)
-				if resourceMonitor != nil {
-					resourceMonitor.LogLatest()
-				}
-
-				ginkgo.By("Deleting the RC")
-				framework.DeleteRCAndWaitForGC(f.ClientSet, f.Namespace.Name, rcName)
-				// Check that the pods really are gone by querying /runningpods on the
-				// node. The /runningpods handler checks the container runtime (or its
-				// cache) and  returns a list of running pods. Some possible causes of
-				// failures are:
-				//   - kubelet deadlock
-				//   - a bug in graceful termination (if it is enabled)
-				//   - docker slow to delete pods (or resource problems causing slowness)
-				start := time.Now()
-				err = waitTillNPodsRunningOnNodes(f.ClientSet, nodeNames, rcName, ns, 0, itArg.timeout)
-				framework.ExpectNoError(err)
-				e2elog.Logf("Deleting %d pods on %d nodes completed in %v after the RC was deleted", totalPods, len(nodeNames),
-					time.Since(start))
-				if resourceMonitor != nil {
-					resourceMonitor.LogCPUSummary()
-				}
+		ginkgo.It("should let Kubelet delete 10 pods per node with in 1 minute", func() {
+			totalPods := 10 * numNodes
+			ginkgo.By(fmt.Sprintf("Creating a RC of %d pods and wait until all pods of this RC are running", totalPods))
+			rcName := fmt.Sprintf("cleanup%d-%s", totalPods, string(uuid.NewUUID()))
+
+			err := framework.RunRC(testutils.RCConfig{
+				Client:       f.ClientSet,
+				Name:         rcName,
+				Namespace:    f.Namespace.Name,
+				Image:        imageutils.GetPauseImageName(),
+				Replicas:     totalPods,
+				NodeSelector: nodeLabels,
 			})
-		}
+			framework.ExpectNoError(err)
+			// Perform a sanity check so that we know all desired pods are
+			// running on the nodes according to kubelet. The timeout is set to
+			// only 30 seconds here because framework.RunRC already waited for all pods to
+			// transition to the running status.
+			err = waitTillNPodsRunningOnNodes(f.ClientSet, nodeNames, rcName, ns, totalPods, time.Second*30)
```

Comment:
I missed that this was called both before and after during the test; the after call has been deleted, the before call is still here.

Can someone from @kubernetes/sig-node-pr-reviews confirm whether this sanity check is necessary for this test? We can't promote it to conformance otherwise

Reply:
I don't know why the second one was removed, but without it this doesn't seem right. 

/hold

Reply:
@timothysc I agree that it doesn't seem right. I'm open to suggestions on a better way to address the concerns in @spiffxp's comment: https://github.com/kubernetes/kubernetes/pull/80731#discussion_r310836102

Reply:
@johnbelamaric do you know any domain experts who could take a look at this?

Reply:
This is the original PR for this test: https://github.com/kubernetes/kubernetes/pull/10313

Reply:
I just checked with @Random-Liu and this seems like a bug to not include the second check. The original PR states that all pods should be checked for deletion after the RC is deleted.

Reply:
To clarify, this test seems to have been written such that it only trusts the kubelet's view of whether pods have been deleted.  Would it be sufficient if it was written to only trust the control-plane's view of whether pods have been deleted?

Reply:
What I was asking be removed were the calls that were asking for the kubelet's view of the world (we don't want to allow conformance tests to rely on the kubelet's API at all).  If they were replaced with calls that asked for the control-plane's view of the world would that be sufficient?

Reply:
This test has been updated to avoid using the Kubelet API in a0469969922f6ad523b1824fdb835fe1512115e1. We've attempted to keep the test's behavior as close to the same otherwise.